inst/odin/lancelot.R

```diff
@@ -1746,16 +1850,36 @@ vaccine_probability_doses[, ] <- min(
   as.numeric(1))
 dim(vaccine_probability_doses) <- c(n_groups, n_doses)
 
-vaccine_attempted_doses[, ] <- vaccine_missed_doses[i, j] + (
-  if (as.integer(step) >= dim(vaccine_dose_step, 3) ||
-      vaccine_n_candidates[i, j] == 0) 0
+tmp_attempted_doses[, ] <- vaccine_missed_doses[i, j] + (
+  if (as.integer(step) >= dim(vaccine_dose_step, 3)) 0
   else vaccine_dose_step[i, j, step + 1])
+dim(tmp_attempted_doses) <- c(n_groups, n_doses)
+
+## Note attempted doses for those competing with a vaccine skip are weighted
+## here, with remaining doses made available to the vaccine skip move below
+vaccine_attempted_doses[, ] <-
+  (if (j == vacc_skip_dose)
+    min(vaccine_n_candidates[i, j] /
+          (vaccine_n_candidates[i, j] +
+             vacc_skip_weight * vacc_skip_n_candidates[i])
+        * tmp_attempted_doses[i, j],
+        vaccine_n_candidates[i, j])
+    else tmp_attempted_doses[i, j])
 dim(vaccine_attempted_doses) <- c(n_groups, n_doses)
 
+vacc_skip_attempted_doses[] <-
```

Comment:
it would be nice to use an extreme case to check if we ever run out of people to vaccinate in the "skip" route whilst not having vaccinated all candidates in the "direct" route.

Reply:
I'm not sure how to test it but it should be mathematically impossible given `vacc_skip_weight <= 1`

Reply:
Let's say there are `x` unwaned candidates, `y` waned candidates, `z` doses to share out between them and the unwaned are weighted `w <= 1` relative to waned.

So we give `w * x * z / (w * x + y)` doses to the unwaned and `y * z/ (w * x + y)` doses to the waned. 

To have an excess of doses for the unwaned while not all waned individuals are vaccinated we need
`w * x * z / (w * x + y) > x` (1)
and
`y * z/ (w * x + y) > y` (2)

Rearranging (2) gives us `z < w * x + y` (3)

Rearranging (1) gives us `z > (w * x + y) / w`. Given w <= 1, then `z > w * x + y` - and since this contradicts (3) the situation is impossible

Reply:
ok! I think you wrote (2) with ">" instead of "<" but the rest of the maths is ok so happy with that thanks @edknock